Sources/Brave/Frontend/Shields/ShieldsViewController.swift

```diff
@@ -317,36 +343,42 @@ class ShieldsViewController: UIViewController, PopoverContentComponent {
   }
 
   @objc private func tappedReportSiteButton() {
-    updateContentView(to: shieldsView.reportBrokenSiteView, animated: true)
+    guard let view = reportViewController?.view else {
+      assertionFailure()
+      return
+    }
+    
+    updateContentView(to: view, animated: true)
   }
 
   @objc private func tappedCancelReportingButton() {
     updateContentView(to: shieldsView.stackView, animated: true)
   }
 
-  @objc private func tappedSubmitReportingButton() {
-    if let url = url {
-      Task { @MainActor in
-        let domain = Domain.getOrCreate(forUrl: url, persistent: !tab.isPrivate)
-        
-        let report = WebcompatReporter.Report(
-          fullUrl: url,
-          areShieldsEnabled: !domain.areAllShieldsOff,
-          adBlockLevel: domain.blockAdsAndTrackingLevel,
-          fingerprintProtectionLevel: domain.finterprintProtectionLevel,
-          adBlockListTitles: FilterListStorage.shared.filterLists.compactMap({ filterList -> String? in
-            guard filterList.isEnabled else { return nil }
-            return filterList.entry.title
-          }),
-          isVPNEnabled: BraveVPN.isConnected
-        )
-        
-        await WebcompatReporter.send(report: report)
-        try await Task.sleep(nanoseconds: NSEC_PER_SEC * 2)
-        guard !self.isBeingDismissed else { return }
-        self.dismiss(animated: true)
-      }
+  @objc private func tappedSubmitReportingButton(url: URL, additionalDetails: String, contactInfo: String) {
+    Task { @MainActor in
+      let domain = Domain.getOrCreate(forUrl: url, persistent: !tab.isPrivate)
+      
+      let report = WebcompatReporter.Report(
+        cleanedURL: url,
+        additionalDetails: additionalDetails,
+        contactInfo: contactInfo,
+        areShieldsEnabled: !domain.areAllShieldsOff,
+        adBlockLevel: domain.blockAdsAndTrackingLevel,
+        fingerprintProtectionLevel: domain.finterprintProtectionLevel,
+        adBlockListTitles: FilterListStorage.shared.filterLists.compactMap({ filterList -> String? in
+          guard filterList.isEnabled else { return nil }
+          return filterList.entry.title
+        }),
+        isVPNEnabled: BraveVPN.isConnected
+      )
+      
+      await WebcompatReporter.send(report: report)
+      try await Task.sleep(nanoseconds: NSEC_PER_SEC * 2)
+      guard !self.isBeingDismissed else { return }
+      self.dismiss(animated: true)
```

Comment:
You're keeping `self` alive here until the task is complete. Better to spawn a timer instead

Reply:
I thought timers in tasks don't work.

Reply:
Since this was moved to the report view I hope this isn't a problem anymore.